src/seedu/addressbook/common/Utils.java

```diff
@@ -34,4 +37,20 @@ public static boolean elementsAreUnique(Collection<?> items) {
         }
         return true;
     }
+
+    /**
+     * Checks if a given string is a valid date.
+     */
+    public static boolean isValidDate(String value) {
+        boolean valid;
+        final String format = "dd-MM-yyyy";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+        try {
+            String parsedDate = LocalDate.parse(value, formatter).format(formatter);
+            valid = value.equals(parsedDate);
+        } catch (DateTimeParseException ex) {
+            valid = false;
+        }
+        return valid;
+    }
```

Comment:
I think the valid can be pulled into the try and catch. So try {return true } catch {return false}

Reply:
Oh I meant to say try {return value.equals(parsedDate)} catch {return false}

Reply:
im extracting variable here